### PR TITLE
Feat(web): Introduce UNSTABLE_Footer component #DS-1367

### PR DESCRIPTION
**Pull request by dlouhak**
### Additional context

Need to be discussed with team:

- [x] unification color design tokens with design
- [x] dynamic columns and its horizontal centering (is it possible to use `grid` or we must use `flex`?)
  - compose with Grid components  
- [x] horizontal alignment of the copyright links (e.g. Jobs aligns links on mobile to the left)
  - user can use utility `text-center` 
- [x] what should be subcomponent and what shouldn't – should the user set himself
  - [x] variable gap in the `Stack` components,
    - let the user handle it
  - [x] inverted color variants in the `Heading`s, `Link`s, ... ?
    - let the user handle it
  - [x] variable gap in the `Grid` components (currently not possible via API) → DS-1388
 - [x] should we decompose component into subcomponent (FooterColumn, FooterHeading, …)?
   - `FooterColumn` would be NTH and should be implemented in the React (DS-1368)
   - consider `FooterColumnItem` as well

### Issue reference

https://jira.almacareer.tech/browse/DS-1367

**TODOs**

- [x] reformat code examples in the README.md
- [x] generate E2E test screenshots

### CODE REVIEW

File: packages/web/src/scss/components/UNSTABLE_Footer/README.md [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+- to achieve the offset between columns and the rest of the Footer, please use spacing utility classes like `mb-*` or `pt-*`.
+
+## Basic Usage
+
+The basic usage of this component could be:
+
+```html
+<footer class="UNSTABLE_Footer">
+  <div class="Container">
+    <div class="Stack Stack--hasIntermediateDividers">
```
Comment (crishpeen) [inline packages/web/src/scss/components/UNSTABLE_Footer/README.md:36]: I thought we wanted to use <Divider />s

File: packages/web/src/scss/components/UNSTABLE_Footer/README.md [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+      <div class="mb-700 mb-tablet-400">
+        <div class="Grid Grid--cols-2 Grid--tablet--cols-4">
+          <nav aria-labelledby="footer-part-one" class="mb-600 mb-tablet-800">
+            <h3 class="typography-body-large-text-bold mb-600" id="footer-part-one">Section headline</h3>
+            <ul class="Stack Stack--hasSpacing" style="--stack-spacing: var(--spirit-space-500)">
+              <li>
+                <a href="www.example.com"> Link </a>
```
Comment (crishpeen) [inline packages/web/src/scss/components/UNSTABLE_Footer/README.md:43]:
```suggestion
                <a href="www.example.com">Link</a>
```
can we try it like this? I don't like the spaces

File: packages/web/src/scss/components/UNSTABLE_Footer/README.md [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+        <li>
+          <a href="www.example.com" class="link-secondary"> Manage cookies </a>
+        </li>
+      </ul>
+    </div>
+  </div>
+</footer>
+```
+
+### Columns variation
```
Comment (crishpeen) [inline packages/web/src/scss/components/UNSTABLE_Footer/README.md:150]: maybe this title and the next paragraph should mention that you are talking only about the "links" section.

File: packages/web/src/scss/components/UNSTABLE_Footer/README.md [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+    </div>
+  </div>
+</footer>
+```
+
+## Color Variants
+
+Currently, Footer comes in two color variants: **cover** (for light
+backgrounds) and **inverted** (for dark backgrounds). Add `Footer--inverted`
+modifier class to apply inverted background color to Footer.
```
Comment (crishpeen) [inline packages/web/src/scss/components/UNSTABLE_Footer/README.md:211]:
```suggestion
modifier class to apply an inverted background color to the Footer.
```

File: packages/web/src/scss/components/UNSTABLE_Footer/README.md [hunk 5]
```diff
@@ -0,0 +1,231 @@
...
+        </li>
+      </ul>
+    </div>
+  </div>
+</footer>
+```
+
+### Columns variation
+
+In case you need specific number of columns for each row, you can use separated `Grid` components.
```
Comment (crishpeen) [inline packages/web/src/scss/components/UNSTABLE_Footer/README.md:152]: In case you need a specific number of columns for each row, you can use separated `Grid` components.

File: packages/web/src/scss/components/UNSTABLE_Footer/_theme.scss [hunk 6]
```diff
@@ -0,0 +1,8 @@
+@use 'sass:map';
+@use '@tokens' as tokens;
+
+$breakpoints: tokens.$breakpoints;
```
Comment (crishpeen) [inline packages/web/src/scss/components/UNSTABLE_Footer/_theme.scss:4]: we can pass only the tablet breakpoint couldn't we?

File: packages/web/src/scss/components/UNSTABLE_Footer/index.html [hunk 7]
```diff
@@ -0,0 +1,1123 @@
+{{#> web/layout/plain }}
+
+<section class="UNSTABLE_Section">
+  <h2 class="docs-Heading">Basic Usage</h2>
+
+  <footer class="UNSTABLE_Footer">
```
Comment (crishpeen) [inline packages/web/src/scss/components/UNSTABLE_Footer/index.html:6]: please add a wrapping stack, because it is part of the documenting component and it will be there in other technologies.

File: packages/web/src/scss/components/UNSTABLE_Footer/index.html [hunk 8]
```diff
@@ -0,0 +1,1123 @@
...
+      </div>
+    </div>
+  </footer>
+
+</section>
+
+<section class="UNSTABLE_Section">
+  <h2 class="docs-Heading">Extended Usage with Section Title as a Link and Nested List</h2>
+
+  <footer class="UNSTABLE_Footer">
```
Comment (crishpeen) [inline packages/web/src/scss/components/UNSTABLE_Footer/index.html:175]: again, please wrap in stack

File: packages/web/src/scss/components/UNSTABLE_Footer/index.html (hunk 7 above)
Comment (dlouhak) [reply packages/web/src/scss/components/UNSTABLE_Footer/index.html:6]: Ok, I didn't think of that because there is only one child

File: packages/web/src/scss/components/UNSTABLE_Footer/README.md [hunk 9]
```diff
@@ -0,0 +1,223 @@
...
+<footer class="UNSTABLE_Footer">
+  <div class="Container">
+    <div class="Grid Grid--cols-2 Grid--tablet--cols-4 mb-700 mb-tablet-400">
+      <nav aria-labelledby="footer-part-one" class="mb-600 mb-tablet-800">
+        <h3 class="typography-body-large-text-bold mb-600" id="footer-part-one">Section headline</h3>
+        <ul class="Stack Stack--hasSpacing" style="--stack-spacing: var(--spirit-space-500)">
+          <li>
+            <a href="www.example.com">Link</a>
+          </li>
+
```
[the remark this answers is not in the document]
Comment (dlouhak) [reply packages/web/src/scss/components/UNSTABLE_Footer/README.md:44]: Personal formatting preference, but I can see that you didn't use it in any other files, so I'll revert it

File: packages/web/src/scss/components/UNSTABLE_Footer/_theme.scss
Comment (vmilersky) [inline packages/web/src/scss/components/UNSTABLE_Footer/_theme.scss:1]: Thought we prefer box-model order (as we do for CSS properties order) for theme variables as well? 🤔

**Review by vmilersky — APPROVED** [review]

File: packages/web/src/scss/components/UNSTABLE_Footer/_theme.scss
Comment (dlouhak) [reply packages/web/src/scss/components/UNSTABLE_Footer/_theme.scss:1]: Not sure if there are any rules how to sort SCSS variables. I would stick to the 🔤.

Comment (vmilersky) [reply packages/web/src/scss/components/UNSTABLE_Footer/_theme.scss:1]: Look into other files, I opened some randomly and found colors usually on the bottom after paddings etc.

Comment (vmilersky) [reply packages/web/src/scss/components/UNSTABLE_Footer/_theme.scss:1]: I don't think we need a rule for everything 😬

Comment (dlouhak) [reply packages/web/src/scss/components/UNSTABLE_Footer/_theme.scss:1]: @crishpeen or @adamkudrna, if you have any preferences, let me know and we can resort them in the https://github.com/lmc-eu/spirit-design-system/pull/1546

File: tests/e2e/demo-homepages.spec.ts-snapshots/web-chromium-linux.png
Comment (crishpeen) [inline tests/e2e/demo-homepages.spec.ts-snapshots/web-chromium-linux.png:1]: you are missing the toggle here

File: packages/web/src/scss/components/UNSTABLE_Footer/index.html [hunk 10]
```diff
@@ -0,0 +1,1026 @@
...
+      <div class="Container">
+        <div class="Grid Grid--cols-2 Grid--tablet--cols-4">
+          <nav aria-labelledby="footer-part-one" class="mb-600 mb-tablet-800">
+            <h3 class="typography-body-large-text-bold mb-600" id="footer-part-one">
+              Section headline
+            </h3>
+
+            <ul class="Stack Stack--hasSpacing" style="--stack-spacing: var(--spirit-space-500)">
+              <li>
+                <a href="www.example.com">
```
Comment (crishpeen) [inline packages/web/src/scss/components/UNSTABLE_Footer/index.html:17]: can you please fix all example.com links to include https? Otherwise, it acts weird.

File: packages/web/src/scss/components/UNSTABLE_Footer/index.html [hunk 11]
```diff
@@ -0,0 +1,1026 @@
...
+
+<section class="UNSTABLE_Section">
+  <h2 class="docs-Heading">Basic Usage</h2>
+  <div class="docs-Stack">
+
+    <footer class="UNSTABLE_Footer">
+      <div class="Container">
+        <div class="Grid Grid--cols-2 Grid--tablet--cols-4">
+          <nav aria-labelledby="footer-part-one" class="mb-600 mb-tablet-800">
+            <h3 class="typography-body-large-text-bold mb-600" id="footer-part-one">
```
Comment (crishpeen) [inline packages/web/src/scss/components/UNSTABLE_Footer/index.html:11]: I wonder if we should make IDs in this section more unique. I know they don't clash now, but they seem too global to me.

File: packages/web/src/scss/components/UNSTABLE_Footer/_theme.scss
Comment (crishpeen) [reply packages/web/src/scss/components/UNSTABLE_Footer/_theme.scss:1]: we try to keep the same order as in CSS properties, but we don't enforce it and don't really care :)) 

In this case, I would probably go: breakpoint-tablet, gap, padding-block, background-color, background-color-inverted. 

But it doesn't really matter and there are only a few variables, so it is well readable anyway.

File: packages/web/src/scss/components/UNSTABLE_Footer/_theme.scss [hunk 12]
```diff
@@ -0,0 +1,8 @@
+@use 'sass:map';
+@use '@tokens' as tokens;
+
+$breakpoint: map.get(tokens.$breakpoints, tablet);
```
Comment (crishpeen) [inline packages/web/src/scss/components/UNSTABLE_Footer/_theme.scss:4]: please $breakpoint-tablet

File: packages/web/src/scss/components/UNSTABLE_Footer/README.md [hunk 13]
```diff
@@ -0,0 +1,204 @@
...
+      <nav aria-labelledby="footer-with-a-lot-of-blocks-part-eight" class="mb-600 mb-tablet-800">…</nav>
+    </div>
+  </div>
+</footer>
+```
+
+## Color Variants
+
+Currently, Footer comes in two color variants: **cover** (for light
```
Comment (crishpeen) [inline packages/web/src/scss/components/UNSTABLE_Footer/README.md:182]: maybe reword this, because I would expect there is also a Footer--cover variant now.

Comment (crishpeen) [reply packages/web/src/scss/components/UNSTABLE_Footer/README.md:182]: Something like:

> If you need to place the Footer on a dark background, add `Footer--inverted` modifier class to the root element.

File: packages/web/src/scss/components/UNSTABLE_Footer/README.md [hunk 14]
```diff
@@ -0,0 +1,204 @@
...
+        <a href="www.example.com" class="link-secondary">Manage cookies</a>
+      </li>
+    </ul>
+  </div>
+</footer>
+```
+
+### Footer Columns Variation
+
+In case you need a specific number of columns for each row in the Footer Columns section with links,
```
Comment (crishpeen) [inline packages/web/src/scss/components/UNSTABLE_Footer/README.md:136]: do we have something called `Footer Columns`?

**Review by crishpeen — COMMENTED** [review]
I don't like the naming (content, utilities, copyright, legal, etc) and don't have time to thoroughly think about it. 🤷 

Otherwise, this looks good. We will see how it works out when actually used before we stabilize it.

File: packages/web/src/scss/components/UNSTABLE_Footer/README.md [hunk 15]
```diff
@@ -0,0 +1,204 @@
...
+  <!-- Footer columns – links relatable to the web hierarchy, etc. -->
+  <!-- Footer content – links to social sites, product logo, etc. -->
+  <!-- Footer copyright – links to the GDPR, cookies, etc. -->
+</footer>
+```
+
+👉 Please, keep in mind that:
+
+- to achieve the desired design, you can use the provided building blocks in any order you need,
+- in case you need dividers between section, use the `UNSTABLE_Divider` component,
```
Comment (curdaj) [inline packages/web/src/scss/components/UNSTABLE_Footer/README.md:26]:
```suggestion
- in case you need dividers between section, use the `Divider` component,
```

File: packages/web/src/scss/components/UNSTABLE_Footer/_theme.scss
Comment (dlouhak) [reply packages/web/src/scss/components/UNSTABLE_Footer/_theme.scss:1]: WIll solve in the #1546

File: packages/web/src/scss/components/UNSTABLE_Footer/index.html (hunk 10 above)
Comment (dlouhak) [reply packages/web/src/scss/components/UNSTABLE_Footer/index.html:17]: WIll solve in the #1546

File: packages/web/src/scss/components/UNSTABLE_Footer/index.html (hunk 11 above)
Comment (dlouhak) [reply packages/web/src/scss/components/UNSTABLE_Footer/index.html:11]: It would clash in the IDE and I try to make all of these IDs as unique as possible. But we can discuss it later in the #1546.

File: packages/web/src/scss/components/UNSTABLE_Footer/_theme.scss (hunk 12 above)
Comment (dlouhak) [reply packages/web/src/scss/components/UNSTABLE_Footer/_theme.scss:4]: WIll solve in the #1546

File: packages/web/src/scss/components/UNSTABLE_Footer/README.md (hunk 14 above)
Comment (dlouhak) [reply packages/web/src/scss/components/UNSTABLE_Footer/README.md:136]: Currently not, but I want to stick with naming above. Maybe it should be `Footer columns`.

Anyway, I would fix it in the #1546.

**Review by crishpeen — APPROVED** [review]
I still don't like the naming and I am afraid that even if we revisit it soon we will leave some remarks of the current one. 

But the rest of the code is ok and we can move on.

**Review by literat — APPROVED** [review]

**Comment by dlouhak** [comment]
We agreed to refactor component `Footer`. Will be refactored with DS-1419.
